test/test_data/env.rs

```diff
@@ -0,0 +1,28 @@
+// Copyright 2021 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#[no_mangle]
+extern "C" {
+    fn __wasilibc_initialize_environ();
```

Comment:
I though we do not need to invoke this directly according to libc doc: https://github.com/WebAssembly/wasi-libc/blob/5ccfab77b097a5d0184f91184952158aa5904c8d/libc-bottom-half/sources/__wasilibc_initialize_environ.c#L8-L17 , but it looks like it doesn't work correctly, and what's more manual execution of `__wasilibc_ensure_environ` does not work. That's why I use `__wasilibc_initialize_environ` here to ensure that the envs are set.

I think this maybe a bug in wasmlibc or wasm-ld. I'm going to report this.